tests/flexmock_test.py

```diff
@@ -49,15 +53,36 @@ def instance_method(self, a):
         pass
 
 
-def assert_raises(exception, method, *kargs, **kwargs):
+@contextmanager
+def assert_raises(expected_exception: Type[BaseException], match: Union[RE_TYPE, str]):
+    """Assert that code raises the correct exception with a correct error message.
+
+    Args:
+        expected_exception: Type of the expected exception.
+        match: String or pattern to match the error message against.
+    """
     try:
-        method(*kargs, **kwargs)
-    except exception:
-        return
-    except Exception:
-        instance = sys.exc_info()[1]
-        print("%s" % instance)
-    raise Exception("%s not raised" % exception.__name__)
+        yield
+    except Exception as raised_exception:
+        if not isinstance(raised_exception, expected_exception):
+            raise AssertionError(
+                f"Expected exception '{type(expected_exception)}' "
+                f"but '{type(raised_exception)}' was raised"
+            ) from raised_exception
+        if match is not None:
```

Comment:
If None is an option it should be in the type, possibly with default value, though I would be in favour of always having to provide a value

Reply:
I'll update the type :ok_hand: I would not add None as default so that if would not encourage skipping message checking. It's better that it needs to be skipped explicitly.

Reply:
https://github.com/flexmock/flexmock/pull/61/commits/e78459d7e453ca715bd14ea41bae751bc2f2f883